newlib/libc/sys/vita/io.c

```diff
@@ -132,6 +132,35 @@ int __vita_release_descriptor(int fd)
 	return 0;
 }
 
+int __vita_duplicate_descriptor(int fd)
+{
+	int fd2 = -1;
+
+	sceKernelLockLwMutex(&_newlib_fd_mutex, 1, 0);
+
+	if (!is_fd_valid(fd))
+	{
+		sceKernelUnlockLwMutex(&_newlib_fd_mutex, 1);
```

Comment:
@DaveeFTW err. You shouldn't really go out of your way to include gotos just for the sake of it. If it fits nicely (like in this case), then they can be useful/look nicer. Generally more helpful when you have more than two failure code-paths, but it still works with two.